Integrations/integration-VMRay.yml

```diff
@@ -0,0 +1,106 @@
+description: Ransomware analysis sandboxing
+configuration:
+- display: Server URL (e.g. https://192.168.0.1)
+  name: server
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: API-Key
+  name: api_key
+  defaultvalue: ""
+  type: 4
+  required: false
+script:
+  script: |+
+    import base64
+    import datetime
+    import os.path
+    import requests
+    import urlparse
+    import time
+    from vmray.rest_api import VMRayRESTAPI, VMRayRESTAPIError
+
+
+    # Get mandantory params
+    # You can use demisto.params()[paramName] or params.paramName to get a specific params.
+    # Params are of the type given in the integration page creation.
+    api_key = demisto.params()['api_key']
+    server = demisto.params()['server']
+
+    api = VMRayRESTAPI(server, api_key)
+
+    url_dict = {
+        "get_results": "/rest/analysis/sample/%s",
+        "get_job_sample": "/rest/job/sample/%s"
+    }
+
+    def submit_sample(api, file):
+        # Submit sample
+        params = {'sample_file': file}
+        data = api.call("POST", "/rest/sample/submit", params)
+        demisto.results(json.dumps(data, indent=2))
+
+    def test_module(api):
+        try:
+            api.call("GET", "/rest/analysis/?_limit=1")
+            demisto.results('ok')
+        except VMRayRESTAPIError as exc:
+            if exc.status_code == 401:
+                demisto.results('Error 401')
+            else:
+                raise
+
+    def get_request(api, url, method='GET'):
+        data = api.call(method, url)
+        demisto.results(data)
+
+    # The command demisto.command() holds the command sent from the user.
+    if demisto.command() == 'test-module':
+        # This is the call made when pressing the integration test button.
+        # demisto.results('ok')
+        test_module(api)
+        sys.exit(0)
+    if demisto.command() == 'upload_sample':
+        # To upload a file the filename is needed. Demisto renames the files
+        # and stores it internaly. To get the filename you need an automation.
+        # You can use demisto.args()[argName] to get a specific arg. args are strings.
+        file = demisto.args()['file_name']
+        demisto.results(submit_sample(api, open(file, "rb")))
+        sys.exit(0)
+    if demisto.command() == 'get_results':
+        job_id = demisto.args()['sample_id']
+        get_request(api, url_dict[demisto.command()] % job_id)
+        sys.exit(0)
```

Comment:
all the sys.exit() can be replaced in a single call at the end of the else-if chain